.github/workflows/commit.yaml

```diff
@@ -82,6 +82,16 @@ jobs:
 
       - run: make test
 
+      - name: "Generate coverage report"  # only once (not per OS)
+        if: runner.os == 'Linux'
+        run: make coverage
+
+      - name: "Upload coverage report"  # only on main push and only once (not per OS)
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main' && runner.os == 'Linux'
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+        run: bash <(curl -s https://codecov.io/bash)
```

Comment:
I executed this locally with the same secret and it worked.

Reply:
I believe this bash uploader is deprecated

https://about.codecov.io/blog/introducing-codecovs-new-uploader/

We should probably just use their action

https://github.com/codecov/codecov-action

Reply:
actually it looks like there are issues for M1 Mac etc. so let's wait until they sort that out. codecov uploads are not critical if they break and I doubt they will really sunset things with the state of support being not great. Thanks for the tip though!